### PR TITLE
Various Sling fixes

### PR DESCRIPTION
**Pull request by ghost**
:cl: steamp0rt
fix: Ascendants should now be able to use hypnosis/gore/gib
fix: Thrall roundend report won't repeat 50 times anymore
tweak: Combined shadowling and thrall roundend report
tweak: Melee-range glare now knocks the target down like a normal stun, instead of paralyzing them.
fix: You can't glare borgs anymore. This wasn't supposed to be possible, but it could apparently be done, and it barfed up runtimes when doing so.
/:cl:

### CODE REVIEW

File: hippiestation/code/datums/antagonists/shadowling.dm [hunk 1]
```diff
@@ -5,6 +5,30 @@
 	antagpanel_category = "Shadowlings"
 	antag_moodlet = /datum/mood_event/sling
 	var/list/objectives_given = list()
+	var/datum/team/shadowling/sling_team
+
+/datum/antagonist/shadowling/create_team(datum/team/shadowling/new_team)
+	if(!new_team)
+		for(var/datum/antagonist/shadowling/H in GLOB.antagonists)
+			if(!H.owner)
```
Comment (Carbonhell) [inline hippiestation/code/datums/antagonists/shadowling.dm:13]: those loops are pretty weird, why don't you just make the team a static variable shared by all shadowling antag datums instead? So you just new it if it is null

File: hippiestation/code/datums/antagonists/shadowling.dm [hunk 2]
```diff
@@ -5,6 +5,30 @@
...
+				return
+		for(var/datum/antagonist/thrall/H in GLOB.antagonists)
+			if(!H.owner)
+				continue
+			if(H.sling_team)
+				sling_team = H.sling_team
+				return
+		sling_team = new /datum/team/shadowling
+		return
+	if(!istype(new_team))
```
Comment (Carbonhell) [inline hippiestation/code/datums/antagonists/shadowling.dm:26]: meh honestly this if check isn't needed

File: hippiestation/code/datums/antagonists/shadowling.dm [hunk 3]
```diff
@@ -52,7 +76,29 @@
...
+		return TRUE
+	return FALSE
+
+/datum/objective/ascend/update_explanation_text()
+	explanation_text = "Ascend to your true form by use of the Ascendance ability. This may only be used with [SSticker.mode.required_thralls] or more collective thralls, while hatched, and is unlocked with the Collective Mind ability."
+
+/mob/living/carbon/human/Stat()
+	. = ..()
+	if(statpanel("Status") && (dna && dna.species) && istype(dna.species, /datum/species/shadow/ling))
+		var/datum/species/shadow/ling/SL = dna.species
```
Comment (Carbonhell) [inline hippiestation/code/datums/antagonists/shadowling.dm:94]: Isn't there any sort of hook inside human Stat? It feels weird to have to redefine human Stat instead

File: hippiestation/code/datums/antagonists/thrall.dm [hunk 4]
```diff
@@ -6,6 +6,30 @@ GLOBAL_LIST_INIT(thrall_spell_types, typecacheof(/obj/effect/proc_holder/spell/s
 	roundend_category = "thralls"
 	antagpanel_category = "Shadowlings"
 	antag_moodlet = /datum/mood_event/thrall
+	var/datum/team/shadowling/sling_team
```
Comment (Carbonhell) [inline hippiestation/code/datums/antagonists/thrall.dm:9]: Wait, why aren't thralls a subtype of shadowling? Aaaaaa

File: hippiestation/code/datums/antagonists/thrall.dm [hunk 5]
```diff
@@ -6,6 +6,30 @@ GLOBAL_LIST_INIT(thrall_spell_types, typecacheof(/obj/effect/proc_holder/spell/s
 	roundend_category = "thralls"
 	antagpanel_category = "Shadowlings"
 	antag_moodlet = /datum/mood_event/thrall
+	var/datum/team/shadowling/sling_team
+
+/datum/antagonist/thrall/create_team(datum/team/shadowling/new_team)
```
Comment (Carbonhell) [inline hippiestation/code/datums/antagonists/thrall.dm:11]: Ugly

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
 
-/obj/effect/proc_holder/spell/proc/shadowling_check(var/mob/living/carbon/human/H)
-	if(!H || !istype(H)) return
-	if(H.dna.species.id == "shadowling" && is_shadow(H)) return TRUE
-	if(H.dna.species.id == "l_shadowling" && is_thrall(H)) return TRUE
-	if(!is_shadow_or_thrall(usr)) to_chat(usr, "<span class='warning'>You can't wrap your head around how to do this.</span>")
-	else if(is_thrall(usr)) to_chat(usr, "<span class='warning'>You aren't powerful enough to do this.</span>")
-	else if(is_shadow(usr)) to_chat(usr, "<span class='warning'>Your telepathic ability is suppressed. Hatch or use Rapid Re-Hatch first.</span>")
-	return 0
+/obj/effect/proc_holder/spell/proc/shadowling_check(var/A)
```
Comment (Carbonhell) [inline hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm]: argument should be mob/living/L rather

**Review by Carbonhell — CHANGES_REQUESTED** [review]

File: hippiestation/code/datums/antagonists/thrall.dm (hunk 4 above)
Comment (ghost) [reply hippiestation/code/datums/antagonists/thrall.dm:9]: Literally every proc would need to be changed for thralls, making it moot point

File: hippiestation/code/datums/antagonists/shadowling.dm (hunk 2 above)
Comment (ghost) [reply hippiestation/code/datums/antagonists/shadowling.dm:26]: This is copy-paste from cult antag datum

File: hippiestation/code/datums/antagonists/shadowling.dm (hunk 1 above)
Comment (ghost) [reply hippiestation/code/datums/antagonists/shadowling.dm:13]: This is copy-paste from cult antag datum. And plus, not a single antag datum has a static team, which leads me to think there's a reason for non-static team vars./

File: hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 6 above)
Comment (ghost) [reply hippiestation/code/game/gamemodes/shadowling/shadowling_abilities.dm]: Done

**Comment by Carbonhell** [comment]
I will handle this PR, dnm till i say it is good

File: hippiestation/code/datums/antagonists/shadowling.dm (hunk 3 above)
Comment (ghost) [reply hippiestation/code/datums/antagonists/shadowling.dm:94]: No, not really. plus I don't think it'll work with a hook

**Review by Carbonhell — APPROVED** [review]

**Comment by Carbonhell** [comment]
I don't like how copypasta it is, but it's true that making thralls a subtype of shadowlings is not feasible for now. The way to collect the team seems really stupid though, since the antagonist list is pretty small i guess it can be ignored but still, it's bad practice. Anyway,not the scope of this PR,merging

**Pull request merged**